main.go

```diff
@@ -8,32 +8,37 @@ import (
 	"os"
 	"strings"
 
-	"github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
 	"sigs.k8s.io/yaml"
 )
 
-func handleDocument(provider string, document []byte) {
-	var release v1alpha1.Release
+func handleDocument(provider, annotationKey string, document []byte) {
+	var release map[string]interface{}
```

Comment:
For a little more type safety, you could use `k8s.io/apimachiner/pkg/apis/meta/v1/unstructured.Unstructured` like this
```go
var release unstructured.Unstructured
err := yaml.UnmarshalStrict(document, &release)
if err != nil {
  // handle
}
annotations := release.GetAnnotations()
// maybe check for `nil` annotations
annotations["key"] = "value"
release.SetAnnotations(annotations)
modified, err := yaml.Marshal(release)
if err != nil {
  // handle
}
```

Reply:
Thank you very much for your input. I've applied your suggestion in https://github.com/giantswarm/kustomize-plugin-releasenotesurlannotationtransformer/pull/2/commits/47a7c0809c80ae8edb7db90c4e550f947e1be2c7